app/telemetry/handler.py

```diff
@@ -19,16 +19,37 @@ def handle_message(event, context):
     :param event: Event payload.
     :param context: Google Cloud Function metadata.
     """
+
+    namespace_key = "document_namespace"
+    doctype_key = "document_type"
+    user_agent_version_key = "user_agent_version"
+
     decompressed = gzip.decompress(base64.b64decode(event["data"])).decode("utf-8")
     telemetry = json.loads(decompressed)
-    if "tiles" in telemetry:
-        for tile in telemetry["tiles"]:
-            if "shim" in tile:
-                ping_adzerk(tile["shim"])
-    elif "metrics" in telemetry:
-        text_metrics = telemetry["metrics"].get("text", {})
-        if "pocket.spoc_shim" in text_metrics:
-            ping_adzerk(text_metrics["pocket.spoc_shim"])
+    attributes = event["attributes"]
+
+    namespace = attributes.get(namespace_key)
+    doctype = attributes.get(doctype_key)
+    user_agent_version = attributes.get(user_agent_version_key)
+
+    if namespace in ["org-mozilla-firefox", "org-mozilla-firefox-beta", "org-mozilla-fenix"]\
+            and "spoc" == doctype:  # Android/Glean
+        if "metrics" in telemetry:
+            text_metrics = telemetry["metrics"].get("text", {})
+            if "pocket.spoc_shim" in text_metrics:
+                ping_adzerk(text_metrics["pocket.spoc_shim"])
+    elif "firefox-desktop" == namespace and "spoc" == doctype:  # Desktop/Glean
+        if int(user_agent_version) >= 122:
```

Comment:
is it guaranteed that desktop >= 122 will always send the new ping and < will always send the old? i'm wondering if the nested if statments have a potential to make scenarios we would be ignoring

Reply:
we chatted on slack, and the change in client telemetry is embedded in the release version, and there isn't a client switch to toggle between behaviors

Reply:
Yeah, for tracking this is the bug where the code was removed
https://bugzilla.mozilla.org/show_bug.cgi?id=1868988